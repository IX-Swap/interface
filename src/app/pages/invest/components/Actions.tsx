import React from 'react'
import { Commitment } from 'types/commitment'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { Button } from '@material-ui/core'
import { CommitmentRoute } from 'app/pages/invest/router/config'

export interface ActionsProps {
  item: Commitment
}

export const Actions = ({ item }: ActionsProps) => {
  return (
    <Button
      component={AppRouterLinkComponent}
      to={CommitmentRoute.view}
      params={{ commitmentId: item._id }}
      size='small'
      variant='outlined'
    >
      View
    </Button>
  )
}
