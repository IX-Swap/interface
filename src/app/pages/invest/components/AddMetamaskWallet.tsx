import { Link, Typography } from '@mui/material'
import { AccountsRoute } from 'app/pages/accounts/router/config'
import React from 'react'

export const AddMetamaskWallet = () => {
  return (
    <>
      <Link
        style={{ backgroundColor: 'transparent', textTransform: 'none' }}
        variant='body1'
        type='button'
        target='_blank'
        href={AccountsRoute.withdrawalAddressesCreate}
      >
        <Typography color='primary'>Add Your Metamask Wallet</Typography>
      </Link>
    </>
  )
}
