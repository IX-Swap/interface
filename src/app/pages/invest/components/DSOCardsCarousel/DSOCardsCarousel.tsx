import React from 'react'
import { CarouselProvider, Slider } from 'pure-react-carousel'
import Box from '@mui/material/Box'
import useStyles from 'app/pages/invest/components/DSOCardsCarousel/DSOCardsCarousel.styles'
import 'pure-react-carousel/dist/react-carousel.es.css'
import { useAppBreakpoints } from 'hooks/useAppBreakpoints'

export interface DSOCardsCarouselProps {
  totalSlides: number
  children: any
}

export const DSOCardsCarousel = (props: DSOCardsCarouselProps) => {
  const classes = useStyles()
  const { isMobile, isTablet } = useAppBreakpoints()
  const { children, totalSlides, ...rest } = props

  const getSlidesCount = () => {
    if (isMobile) return 1.1
    if (isTablet) return 1.6
    return 2.4
  }

  return (
    <div {...rest} className={classes.container}>
      <CarouselProvider
        naturalSlideWidth={320}
        naturalSlideHeight={400}
        visibleSlides={getSlidesCount()}
        totalSlides={totalSlides}
        step={1}
        isIntrinsicHeight
      >
        <Box className={classes.wrapper}>
          <Slider>{children}</Slider>
        </Box>
      </CarouselProvider>
    </div>
  )
}
