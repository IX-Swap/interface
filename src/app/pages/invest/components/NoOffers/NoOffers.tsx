import React from 'react'
import { Grid, Typography } from '@mui/material'
import { ReactComponent as MoonIcon } from './image/moon.svg'

export const NoOffers = () => (
  <Grid
    container
    justifyContent={'center'}
    alignItems={'center'}
    direction={'column'}
    sx={theme => ({
      border: `1px dashed ${theme.palette.button.borderOutlined}`,
      py: theme.spacing(3),
      borderRadius: 2
    })}
  >
    <Grid item>
      <MoonIcon />
    </Grid>
    <Grid item mt={1.5}>
      <Typography variant={'h5'}>No offers yet</Typography>
    </Grid>
    <Grid item mt={1}>
      <Typography color={'text.secondary'} fontWeight={400}>
        No offers yet, please check back later
      </Typography>
    </Grid>
  </Grid>
)
