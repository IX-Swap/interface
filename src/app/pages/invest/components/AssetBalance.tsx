import React from 'react'
import { useBalancesByAssetId } from 'hooks/balance/useBalancesByAssetId'
import { LabelledValue } from 'components/LabelledValue'
import { formatMoney } from 'helpers/numbers'

export interface AssetBalanceProps {
  assetId: string
  symbol: string
}

export const AssetBalance = (props: AssetBalanceProps) => {
  const { assetId, symbol } = props
  const { data, isLoading, isIdle } = useBalancesByAssetId(assetId)

  if (isLoading || isIdle) {
    return null
  }

  return (
    <LabelledValue
      label='Account Balance'
      value={formatMoney(
        data.map[assetId]?.available ?? 0,
        data.map[assetId]?.symbol ?? symbol
      )}
    />
  )
}
