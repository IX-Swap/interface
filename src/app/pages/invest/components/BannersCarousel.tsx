import React from 'react'
import { Box, Grid, Typography } from '@mui/material'
import { useTheme } from '@mui/material/styles'
import { useBannersList } from 'app/pages/admin/hooks/useBannersList'
import { Avatar } from 'components/Avatar'
import { Carousel } from 'react-responsive-carousel'
import 'react-responsive-carousel/lib/styles/carousel.min.css'
import { DataroomBanner } from 'types/dataroomBanner'

export const BannersCarousel = () => {
  const theme = useTheme()
  const { data, isLoading } = useBannersList()

  if (typeof data === 'undefined' || isLoading) {
    return null
  }

  return (
    <Grid
      style={{ paddingTop: theme.spacing(3), paddingBottom: theme.spacing(4) }}
    >
      <Carousel
        showThumbs={false}
        showArrows={false}
        showStatus={false}
        showIndicators={false}
        interval={3000}
        infiniteLoop
        autoPlay
      >
        {data.map((item: DataroomBanner) => {
          const { title, _id } = item

          return (
            <div key={_id}>
              {title !== '' ? (
                <Typography
                  variant='h4'
                  style={{
                    marginBottom: theme.spacing(2.5),
                    textAlign: 'left'
                  }}
                >
                  {title}
                </Typography>
              ) : null}
              <Avatar
                size={['100%', 'auto']}
                documentId={_id}
                variant={'square'}
                type={'banner'}
                fallback={<Box />}
              />
            </div>
          )
        })}
      </Carousel>
    </Grid>
  )
}
