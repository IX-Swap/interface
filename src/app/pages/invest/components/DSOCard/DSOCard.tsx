import React from 'react'
import { DigitalSecurityOffering } from 'types/dso'
import Card from '@mui/material/Card'
import useStyles from './DSOCard.styles'
import { Button, Grid } from '@mui/material'
import { DSOCardAction } from 'app/pages/invest/components/DSOCard/DSOCardAction'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { DSOCardCover } from 'app/pages/invest/components/DSOCard/DSOCardCover'
import { DSOLogo } from 'app/components/DSO/components/DSOLogo'
import { PrimaryCardContent } from 'app/pages/invest/components/DSOCard/PrimaryCardContent'
import { OTCCardContent } from 'app/pages/invest/components/DSOCard/OTCCardContent'

export interface DSOCardProps {
  type: 'Primary' | 'OTC' | 'TopOffers'
  data: DigitalSecurityOffering
  viewURL: string
}

export const DSOCard = (props: DSOCardProps) => {
  const classes = useStyles()
  const typeWithLogo = ['OTC', 'TopOffers']

  const { data, viewURL, type } = props

  return (
    <Card
      data-testid='primaryDsoCard'
      className={classes.container}
      variant='outlined'
      elevation={0}
      style={{ flexDirection: 'column' }}
    >
      {type !== 'OTC' && (
        <DSOCardCover type={type} data={data} viewURL={viewURL} />
      )}

      <Grid
        container
        justifyContent={'center'}
        alignItems={'center'}
        className={classes.logoWrapper}
      >
        <Grid item>
          <DSOLogo
            uri={typeWithLogo.includes(type) ? '/dataroom/raw/' : undefined}
            size={80}
            dsoId={type === 'Primary' ? data._id : data.logo}
            variant='circular'
            className={classes.logo}
          />
        </Grid>
      </Grid>

      {type === 'OTC' ? (
        <OTCCardContent data={data} />
      ) : (
        <PrimaryCardContent data={data} />
      )}

      {/* TODO Uncomment and add value from api response when backend part will be completed */}
      {/* <Typography className={classes.status}> */}
      {/*  Accredited Investors only */}
      {/* </Typography> */}

      <Grid container item justifyContent='space-between' alignItems={'center'}>
        <DSOCardAction type={type} data={data} />
        {type !== 'OTC' ? (
          <Button
            className={classes.link}
            component={AppRouterLinkComponent}
            to={viewURL}
            params={{ dsoId: data._id, issuerId: data.createdBy }}
            variant='text'
            color='primary'
          >
            Learn More
          </Button>
        ) : null}
      </Grid>
    </Card>
  )
}
