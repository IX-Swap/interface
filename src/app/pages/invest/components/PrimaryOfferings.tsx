import React from 'react'
import { InvestRoute } from 'app/pages/invest/router/config'
import { Button, Grid, Typography } from '@mui/material'
import { useStyles } from 'app/pages/invest/components/styles/OTCMarket.style'
import { useTableWithPagination } from 'components/TableWithPagination/hooks/useTableWithPagination'
import { dsoQueryKeys } from 'config/queryKeys'
import { issuanceURL } from 'config/apiURL'
import { DigitalSecurityOffering } from 'types/dso'
import { useQueryFilter } from 'hooks/filters/useQueryFilter'
import { DSOCard } from 'app/pages/invest/components/DSOCard/DSOCard'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { Count } from 'app/pages/invest/components/Count'

export const PrimaryOfferings = () => {
  const classes = useStyles()
  const { getFilterValue } = useQueryFilter()
  const search = getFilterValue('search')
  const primaryOfferingSearch = getFilterValue('primaryOfferingSearch')

  const { items, status, total } = useTableWithPagination({
    queryKey: dsoQueryKeys.getPromoted,
    uri: issuanceURL.dso.getAllPromoted,
    defaultFilter: { search: search ?? primaryOfferingSearch },
    queryEnabled: true,
    defaultRowsPerPage: 5,
    disabledUseEffect: true
  })

  if (status === 'loading' || items.length === undefined) {
    return null
  }

  return (
    <Grid container direction={'column'} spacing={3}>
      <Grid
        item
        container
        alignItems={'center'}
        justifyContent={'space-between'}
      >
        <Grid item>
          <Typography
            variant='h4'
            display={'inline-flex'}
            alignItems={'center'}
          >
            Primary Offerings <Count value={total} />
          </Typography>
        </Grid>
        <Grid item>
          <Button
            component={AppRouterLinkComponent}
            color='primary'
            variant='text'
            to={InvestRoute.landing}
            data-testid='invest-link'
            sx={{ px: 0 }}
          >
            View all
          </Button>
        </Grid>
      </Grid>

      <Grid container item wrap={'wrap'} className={classes.container}>
        {(items as DigitalSecurityOffering[]).slice(0, 3).map((dso, i) => (
          <DSOCard
            type={'Primary'}
            data={dso}
            viewURL={InvestRoute.view}
            key={dso._id}
          />
        ))}
      </Grid>
    </Grid>
  )
}
