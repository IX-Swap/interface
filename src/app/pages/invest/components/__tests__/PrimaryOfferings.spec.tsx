import * as useStyles from 'app/pages/invest/components/styles/OTCMarket.style'
import * as useTableWithPagination from 'components/TableWithPagination/hooks/useTableWithPagination'
import * as useQueryFilter from 'hooks/filters/useQueryFilter'
import { render } from '@testing-library/react'
import React from 'react'
import { dso } from '__fixtures__/issuance'
import { InvestRoute } from 'app/pages/invest/router/config'
import { DSOCard } from 'app/pages/invest/components/DSOCard/DSOCard'
import { PrimaryOfferings } from 'app/pages/invest/components/PrimaryOfferings'
import { NoOffers } from 'app/pages/invest/components/NoOffers/NoOffers'
import { DSOCardsCarousel } from 'app/pages/invest/components/DSOCardsCarousel/DSOCardsCarousel'
import * as useAppBreakpoints from 'hooks/useAppBreakpoints'

jest.mock('app/pages/invest/components/DSOCard/DSOCard', () => ({
  DSOCard: jest.fn(() => null)
}))

jest.mock('app/pages/invest/components/NoOffers/NoOffers', () => ({
  NoOffers: jest.fn(() => null)
}))

jest.mock(
  'app/pages/invest/components/DSOCardsCarousel/DSOCardsCarousel',
  () => ({
    DSOCardsCarousel: jest.fn(() => null)
  })
)

jest.mock('@mui/material/Typography', () => jest.fn(() => null))

describe('PrimaryOfferings', () => {
  const getFilterValueFn = jest.fn(() => 'search')
  const objSuccessfulResponse = {
    total: 1,
    status: undefined,
    items: [dso]
  }

  const objLoadingResponse = {
    total: 1,
    status: 'loading',
    items: [dso]
  }

  const objEmptyResponse = {
    total: 0,
    status: undefined,
    items: []
  }

  beforeEach(() => {
    jest
      .spyOn(useQueryFilter, 'useQueryFilter')
      .mockImplementation(() => ({ getFilterValue: getFilterValueFn } as any))
    jest
      .spyOn(useStyles, 'useStyles')
      .mockReturnValueOnce({ container: 'abc' } as any)
  })

  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('renders NoOffers component when data is empty array', () => {
    jest
      .spyOn(useTableWithPagination, 'useTableWithPagination')
      .mockImplementation(() => objEmptyResponse as any)

    render(<PrimaryOfferings />)

    expect(NoOffers).toHaveBeenCalledTimes(1)
  })

  it('renders DSOCard component with correct props, should match snapshot', () => {
    jest
      .spyOn(useTableWithPagination, 'useTableWithPagination')
      .mockImplementation(() => objSuccessfulResponse as any)

    const { container } = render(<PrimaryOfferings />)

    expect(DSOCard).toHaveBeenCalledWith(
      expect.objectContaining({
        type: 'Primary',
        data: dso,
        viewURL: InvestRoute.view
      }),
      {}
    )
    expect(container).toMatchSnapshot()
  })

  it('renders DSOCardsCarousel component with correct props when isMiniLaptop is true', () => {
    jest.spyOn(useAppBreakpoints, 'useAppBreakpoints').mockReturnValueOnce({
      isMiniLaptop: true
    } as any)

    jest
      .spyOn(useTableWithPagination, 'useTableWithPagination')
      .mockImplementation(() => objSuccessfulResponse as any)

    const { container } = render(<PrimaryOfferings />)

    expect(DSOCardsCarousel).toHaveBeenCalledWith(
      expect.objectContaining({
        totalSlides: objSuccessfulResponse.items.length
      }),
      {}
    )
  })
})
