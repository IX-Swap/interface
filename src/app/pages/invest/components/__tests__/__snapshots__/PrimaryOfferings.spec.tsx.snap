// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryOfferings renders DSOCard component with correct props, should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-32q0xd-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1gb55do-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container abc css-11lq3yg-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
